data_structures/binary_tree/maximum_fenwick_tree.py

```diff
@@ -0,0 +1,50 @@
+class MaxFenwickTree:
+    """
+    Maximum Fenwick Tree
+    ---------
+    >>> ft = MaxFenwickTree(5)
+    >>> ft.query(0, 5)
+    0
+    >>> ft.update(2, 20)
+    >>> ft.query(0, 5)
+    20
+    >>> ft.update(5, 10)
+    >>> ft.query(2, 5)
+    10
+    >>> ft.update(2, 0)
+    >>> ft.query(0, 5)
+    10
+    """
+
+    def __init__(self, size: int) -> None:  # Create Fenwick tree with specified size
+        self.size = size
+        self.arr = [0] * (size + 1)
+        self.tree = [0] * (size + 1)
+
+    def update(
+        self, index: int, value: int
+    ) -> None:  # Set index (1-Based) to value in O(lg^2 N)
```

Comment:
Do not wrap a line just for a comment.  Instead, put the comment on a separate line or in a docstring.